go-controller/pkg/node/gateway_shared_intf.go

```diff
@@ -1154,7 +1154,7 @@ func (npw *nodePortWatcher) DeleteEndpointSlice(epSlice *discovery.EndpointSlice
 	klog.V(5).Infof("Deleting endpointslice %s in namespace %s", epSlice.Name, epSlice.Namespace)
 	// remove rules for endpoints and add back normal ones
 	namespacedName, err := util.ServiceFromEndpointSlice(epSlice, netInfo)
-	if err != nil {
+	if err != nil || namespacedName == nil {
```

Comment:
Is having an empty serviceName on label key an error for us? Or we should ignore the processing for such cases with a warning..and return nil ? meaning the upstream test that causes this I wonder what happens there if we start returning an error

Reply:
what I meant was shouldn't we return nil here if namesacedName is nil? why do we return error?

Reply:
oh shit :) bad semantics,, from the actual func we return nil and so here err=nil silly me